vcu/Src/motorController.c

```diff
@@ -45,302 +25,180 @@ float min(float a, float b)
-    {
-        sendThrottleValueToMCs(0, 0);
-        vTaskDelay(pdMS_TO_TICKS(THROTTLE_POLL_TIME_MS));
-    }
+    // Todo - disable unused broadcasts
 
-    if ((StateInverterRight & INVERTER_STATE_MASK) != INVERTER_BRIDGE_DISABLED) {
-        ERROR_PRINT("Timeout waiting for MC Right to be ready to turn on\n");
-        ERROR_PRINT("StateInverterRight %d\n", (uint8_t)(StateInverterRight & INVERTER_STATE_MASK));
-        return HAL_TIMEOUT;
-    }
+    // This isn't really used, but may have future use?
+    initMotorControllerSettings();
 
-    DEBUG_PRINT("Right motor controller ready.\n");
+    requestTorqueFromMC(0,0);
+    motors_active = true;
 
-    DEBUG_PRINT("Initializing default settings...");
-    initMotorControllerProcanSettings();
+    return HAL_OK;
+}
 
-    if (mcLeftCommand(INVERTER_ENABLE_BRIDGE) != HAL_OK) {
-        ERROR_PRINT("Failed to send enable bridge command to MC Left");
-        return HAL_ERROR;
-    }
+HAL_StatusTypeDef sendDisableMC(void) {
+    VCU_INV_Inverter_Enable = INVERTER_OFF;
+    VCU_INV_Inverter_Discharge = INVERTER_DISCHARGE_DISABLE;
+    VCU_INV_Speed_Mode_Enable = SPEED_MODE_OVERRIDE_FALSE;
+    VCU_INV_Torque_Limit_Command = TORQUE_LIMIT_OVERRIDE_FALSE;
 
-    if (mcRightCommand(INVERTER_ENABLE_BRIDGE) != HAL_OK) {
-        ERROR_PRINT("Failed to send enable bridge command to MC right");
+    if (sendCAN_MC_Command_Message() != HAL_OK) {
+        ERROR_PRINT("Failed to send disable message to MC\n");
         return HAL_ERROR;
     }
-    motors_active = true;
 
     return HAL_OK;
 }
 
-HAL_StatusTypeDef mcShutdown()
-{
+HAL_StatusTypeDef mcDisable() {   
+    // todo addison
 	motors_active = false;
 	// Wait for final throttle messages to exit our CAN queue
     vTaskDelay(pdMS_TO_TICKS(250));
-    if (mcLeftCommand(INVERTER_DISABLE_BRIDGE) != HAL_OK) {
-        ERROR_PRINT("Failed to send init disable bridge command to MC Left");
-        return HAL_ERROR;
-    }
-
-    if (mcRightCommand(INVERTER_DISABLE_BRIDGE) != HAL_OK) {
-        ERROR_PRINT("Failed to send init disable bridge command to MC Right");
-        return HAL_ERROR;
-    }
-
-    return HAL_OK;
+    
+    return sendDisableMC();
 }
 
-float map_range_float(float in, float low, float high, float low_out, float high_out) {
-    if (in < low) {
-        in = low;
-    } else if (in > high) {
-        in = high;
-    }
-    float in_range = high - low;
-    float out_range = high_out - low_out;
 
-    return (in - low) * out_range / in_range + low_out;
-}
+HAL_StatusTypeDef sendLockoutReleaseToMC() {
+    // Based on Cascadia Motion documentation, need to send an inverter disable command to release lockout
+    // Note - lockout will not disable if inverter is faulted
 
-float limit(float in, float min, float max)
-{
-    if (in > max) {
-        in = max;
-    } else if (in < min) {
-        in = min;
+    if (isLockoutDisabled()) { // disabled
+        // Don't need to release
```

Comment:
Argument could be made that it can not hurt to just send the disable message again